sys.path test failures with IPython 7.2.0 (ipykrenel 5.1.0, 4.10.0, 4.8.2)

We've started to get build failures in Fedora recently and I don't know what's wrong. It started to happening on 4.8.2 ([full build log](https://github.com/ipython/ipykernel/files/2853653/build-4.8.2.log) with nosetests). Since the assertions are not very useful I've switched to pytest as you do on Travis CI (although README still suggests nosetests). This is the build failure with pytest and 4.10.0:

```
+ /usr/bin/python3 -m pytest ipykernel
============================= test session starts ==============================
platform linux -- Python 3.7.2, pytest-3.9.3, py-1.7.0, pluggy-0.8.0
rootdir: /builddir/build/BUILD/ipykernel-4.10.0, inifile:
collected 95 items
ipykernel/inprocess/tests/test_kernel.py .s..                            [  4%]
ipykernel/inprocess/tests/test_kernelmanager.py .....                    [  9%]
ipykernel/tests/test_connect.py ..                                       [ 11%]
ipykernel/tests/test_embed_kernel.py ...                                 [ 14%]
ipykernel/tests/test_eventloop.py .                                      [ 15%]
ipykernel/tests/test_io.py .                                             [ 16%]
ipykernel/tests/test_jsonutil.py .....                                   [ 22%]
ipykernel/tests/test_kernel.py .FF....s....Fs.                           [ 37%]
ipykernel/tests/test_kernelspec.py ..........                            [ 48%]
ipykernel/tests/test_message_spec.py ......................              [ 71%]
ipykernel/tests/test_pickleutil.py .....                                 [ 76%]
ipykernel/tests/test_serialize.py ..............                         [ 91%]
ipykernel/tests/test_start_kernel.py ..                                  [ 93%]
ipykernel/tests/test_zmq_shell.py ......                                 [100%]
=================================== FAILURES ===================================
________________________________ test_sys_path _________________________________
    def test_sys_path():
        """test that sys.path doesn't get messed up by default"""
        with kernel() as kc:
            msg_id, content = execute(kc=kc, code="import sys; print (repr(sys.path[0]))")
            stdout, stderr = assemble_output(kc.iopub_channel)
>           assert stdout == "''\n"
E           assert "'/builddir/b...nel-4.10.0'\n" == "''\n"
E             - '/builddir/build/BUILD/ipykernel-4.10.0'
E             + ''
ipykernel/tests/test_kernel.py:56: AssertionError
__________________________ test_sys_path_profile_dir ___________________________
    def test_sys_path_profile_dir():
        """test that sys.path doesn't get messed up when `--profile-dir` is specified"""
    
        with new_kernel(['--profile-dir', locate_profile('default')]) as kc:
            msg_id, content = execute(kc=kc, code="import sys; print (repr(sys.path[0]))")
            stdout, stderr = assemble_output(kc.iopub_channel)
>           assert stdout == "''\n"
E           assert "'/builddir/b...nel-4.10.0'\n" == "''\n"
E             - '/builddir/build/BUILD/ipykernel-4.10.0'
E             + ''
ipykernel/tests/test_kernel.py:64: AssertionError
________________________________ test_complete _________________________________
    def test_complete():
        with kernel() as kc:
            execute(u'a = 1', kc=kc)
            wait_for_idle(kc)
            cell = 'import IPython\nb = a.'
            kc.complete(cell)
            reply = kc.get_shell_msg(block=True, timeout=TIMEOUT)
            c = reply['content']
            assert c['status'] == 'ok'
>           assert c['cursor_start'] == cell.find('a.')
E           AssertionError: assert 21 == 19
E            +  where 19 = <built-in method find of str object at 0x7f7a3ba748e8>('a.')
E            +    where <built-in method find of str object at 0x7f7a3ba748e8> = 'import IPython\nb = a.'.find
ipykernel/tests/test_kernel.py:252: AssertionError
=============================== warnings summary ===============================
...
========= 3 failed, 89 passed, 3 skipped, 50 warnings in 35.52 seconds =========
```

Where `/builddir/build/BUILD/ipykernel-4.10.0` is pwd. [Full build.log](https://github.com/ipython/ipykernel/files/2853656/build.log)

With 5.1.0, the failures are close to identical:

```
________________________________ test_sys_path _________________________________

    def test_sys_path():
        """test that sys.path doesn't get messed up by default"""
        with kernel() as kc:
            msg_id, content = execute(kc=kc, code="import sys; print (repr(sys.path[0]))")
            stdout, stderr = assemble_output(kc.iopub_channel)
>           assert stdout == "''\n"
E           assert "'/builddir/b...rnel-5.1.0'\n" == "''\n"
E             - '/builddir/build/BUILD/ipykernel-5.1.0'
E             + ''

ipykernel/tests/test_kernel.py:57: AssertionError
__________________________ test_sys_path_profile_dir ___________________________

    def test_sys_path_profile_dir():
        """test that sys.path doesn't get messed up when `--profile-dir` is specified"""
    
        with new_kernel(['--profile-dir', locate_profile('default')]) as kc:
            msg_id, content = execute(kc=kc, code="import sys; print (repr(sys.path[0]))")
            stdout, stderr = assemble_output(kc.iopub_channel)
>           assert stdout == "''\n"
E           assert "'/builddir/b...rnel-5.1.0'\n" == "''\n"
E             - '/builddir/build/BUILD/ipykernel-5.1.0'
E             + ''

ipykernel/tests/test_kernel.py:65: AssertionError
________________________________ test_complete _________________________________

    def test_complete():
        with kernel() as kc:
            execute(u'a = 1', kc=kc)
            wait_for_idle(kc)
            cell = 'import IPython\nb = a.'
            kc.complete(cell)
            reply = kc.get_shell_msg(block=True, timeout=TIMEOUT)
            c = reply['content']
            assert c['status'] == 'ok'
>           assert c['cursor_start'] == cell.find('a.')
E           AssertionError: assert 21 == 19
E            +  where 19 = <built-in method find of str object at 0x7f972d780df8>('a.')
E            +    where <built-in method find of str object at 0x7f972d780df8> = 'import IPython\nb = a.'.find

ipykernel/tests/test_kernel.py:253: AssertionError
```

Again, `/builddir/build/BUILD/ipykernel-5.1.0` is pwd.

[Full build.log with pip freeze call](https://github.com/ipython/ipykernel/files/2853669/build.log)

This is from the latest run:

```
+ /usr/bin/python3 -m pip freeze
alabaster==0.7.12
atomicwrites==1.1.5
attrs==17.4.0
Babel==2.6.0
backcall==0.1.0
chardet==3.0.4
decorator==4.3.0
docutils==0.14
idna==2.7
imagesize==1.0.0
ipykernel==5.1.0
ipython==7.2.0
ipython-genutils==0.1.0
jedi==0.12.1
Jinja2==2.10
jupyter-client==5.2.3
jupyter-core==4.4.0
MarkupSafe==1.0
mock==2.0.0
more-itertools==4.1.0
nose==1.3.7
nose-warnings-filters==0.1.5
numpy==1.16.1
packaging==17.1
parso==0.3.1
pbr==4.2.0
pexpect==4.6.0
pickleshare==0.7.4
pluggy==0.8.0
prompt-toolkit==2.0.7
ptyprocess==0.6.0
py==1.7.0
pycurl==7.43.0.2
Pygments==2.2.0
pyparsing==2.3.0
PySocks==1.6.8
pytest==3.9.3
python-dateutil==2.7.5
pytz==2018.5
pyzmq==17.0.0
requests==2.21.0
simplegeneric==0.8.1
six==1.11.0
snowballstemmer==1.2.1
Sphinx==1.7.6
sphinx-rtd-theme==0.4.2
sphinxcontrib-github-alt==1.0
sphinxcontrib-websupport==1.1.0.dev20180715
SQLAlchemy==1.2.17
tornado==5.0.2
traitlets==4.3.2
urllib3==1.24.1
wcwidth==0.1.7
Whoosh==2.7.4
```


I've noticed that the package fails to build for quite a while now and I cannot longer trace the particular change in the buildroot that started this :(

Is it some kind of version mismatch? I seriously don't know why worked before, but doesn't now.
Thanks for any pointers.

--- a/.travis.yml
+++ b/.travis.yml
@@ -10,8 +10,7 @@ install:
   - |
     # pip install
     pip install --upgrade setuptools pip
-    pip install --pre .
-    pip install ipykernel[test] codecov
+    pip install --pre --upgrade --upgrade-strategy=eager .[test] codecov
   - |
     # install matplotlib
     if [[ "$TRAVIS_PYTHON_VERSION" == "3.6" ]]; then

--- a/ipykernel/kernelbase.py
+++ b/ipykernel/kernelbase.py
@@ -795,7 +795,6 @@ class Kernel(SingletonConfigurable):
         yield gen.sleep(self.stop_on_error_timeout)
         self._aborting = False
 
-    @gen.coroutine
     def _send_abort_reply(self, stream, msg, idents):
         """Send a reply to an aborted request"""
         self.log.info("Aborting:")

--- a/ipykernel/tests/test_kernel.py
+++ b/ipykernel/tests/test_kernel.py
@@ -4,6 +4,7 @@
 # Copyright (c) IPython Development Team.
 # Distributed under the terms of the Modified BSD License.
 
+import ast
 import io
 import os.path
 import sys
@@ -52,17 +53,27 @@ def test_simple_print():
 def test_sys_path():
     """test that sys.path doesn't get messed up by default"""
     with kernel() as kc:
-        msg_id, content = execute(kc=kc, code="import sys; print (repr(sys.path[0]))")
+        msg_id, content = execute(kc=kc, code="import sys; print(repr(sys.path))")
         stdout, stderr = assemble_output(kc.iopub_channel)
-        assert stdout == "''\n"
+    # for error-output on failure
+    sys.stderr.write(stderr)
+
+    sys_path = ast.literal_eval(stdout.strip())
+    assert '' in sys_path
+
 
 def test_sys_path_profile_dir():
     """test that sys.path doesn't get messed up when `--profile-dir` is specified"""
 
     with new_kernel(['--profile-dir', locate_profile('default')]) as kc:
-        msg_id, content = execute(kc=kc, code="import sys; print (repr(sys.path[0]))")
+        msg_id, content = execute(kc=kc, code="import sys; print(repr(sys.path))")
         stdout, stderr = assemble_output(kc.iopub_channel)
-        assert stdout == "''\n"
+    # for error-output on failure
+    sys.stderr.write(stderr)
+
+    sys_path = ast.literal_eval(stdout.strip())
+    assert '' in sys_path
+
 
 @dec.skipif(sys.platform == 'win32', "subprocess prints fail on Windows")
 def test_subprocess_print():
@@ -248,14 +259,23 @@ def test_complete():
         cell = 'import IPython\nb = a.'
         kc.complete(cell)
         reply = kc.get_shell_msg(block=True, timeout=TIMEOUT)
-        c = reply['content']
-        assert c['status'] == 'ok'
-        assert c['cursor_start'] == cell.find('a.')
-        assert c['cursor_end'] == cell.find('a.') + 2
-        matches = c['matches']
-        nt.assert_greater(len(matches), 0)
-        for match in matches:
-            assert match[:2] == 'a.'
+
+    c = reply['content']
+    assert c['status'] == 'ok'
+    start = cell.find('a.')
+    end = start + 2
+    assert c['cursor_end'] == cell.find('a.') + 2
+    assert c['cursor_start'] <= end
+
+    # there are many right answers for cursor_start,
+    # so verify application of the completion
+    # rather than the value of cursor_start
+
+    matches = c['matches']
+    assert matches
+    for m in matches:
+        completed = cell[:c['cursor_start']] + m
+        assert completed.startswith(cell)
 
 
 @dec.skip_without('matplotlib')

